external/fetcher.go

```diff
@@ -64,7 +64,7 @@ func DownloadStream(r io.ReadCloser, checksum, localPath string) error {
 	if err != nil {
 		return err
 	}
-	file, err := os.Create(localPath)
+	file, err := os.OpenFile(localPath, os.O_RDWR|os.O_CREATE|os.O_TRUNC, 0700)
```

Comment:
Create doesn't by default give execute permissions (even to owner)

Reply:
We should pass in the permissions or leave it up to the caller to set them after, 0700 does not seem like a normal permission set for a function called simply called DownloadStream

Reply:
so after some thought I added a chmod in the execution step for artifacts, but not local files. I figured changing permissions on local files was not something we wanted to do in this api.